core/src/main/java/org/apache/metamodel/query/parser/FromItemParser.java

```diff
@@ -66,7 +71,11 @@ public void parse(String delim, String itemToken) {
                 fromItem.setAlias(alias);
             }
         } else if (itemToken.toUpperCase().indexOf(" JOIN ") != -1) {
-            fromItem = parseJoinItem(itemToken);
+            if(itemToken.split("(?i) JOIN ").length == 2) {
```

Comment:
I'm thinking that instead of splitting this text twice (here and in parseAllJoinItems) then wouldn't it be better to not have this if-block and just always apply parseAllJoinItems(...)?